src/triton_fil/triton_tensor.cuh

```diff
@@ -102,15 +102,14 @@ class TritonTensor {
   TritonTensor(
       std::vector<RawOutputBuffer>&& buffers, const std::string& name,
       const std::vector<int64_t>& shape, const TRITONSERVER_DataType dtype,
-      raft::handle_t& handle)
+      TRITONSERVER_MemoryType target_memory, raft::handle_t* handle)
       : name_{name}, shape_{shape}, dtype_{dtype_},
         size_bytes_{sizeof(T) * product(shape_)},
         is_owner_{
             buffers.size() > 1  // non-contiguous
-            || buffers[0].memory_type !=
-                   TRITONSERVER_MEMORY_GPU  // non-device memory
+            || buffers[0].memory_type != target_memory
         },
-        stream_{handle.get_stream()}, buffer{[&] {
+        stream_{(handle ? handle->get_stream() : 0)}, buffer{[&] {
           if (is_owner_) {
             return allocate_device_memory<non_const_T>(
```

Comment:
Just noticed that we are always allocating device memory rather than host memory. I'm surprised this is working, given that? Or maybe I'm missing something?

Reply:
This is fine, since the data arrives on the host memory and the desired destination is also in the host memory, and thus `is_owner_` is false always.

Reply:
Ah, I see! We should double-check that it is not possible for the data to arrive on the GPU for models with `kind` CPU, but I think you're right now that I think about it. I'll check that out myself as I'm looking at this PR.

Even if that is not possible, we should consider adding the capability of handling that situation and controlling GPU/CPU execution via another flag in order to allow for more flexibility in deployments, but that should *definitely* wait for a later PR. I'll open an issue if necessary.